sorting/merge_sort.c

```diff
@@ -101,18 +106,32 @@ void merge_sort(int *a, int n, int l, int r)
 int main(void)
 {
     int *a, n, i;
+    printf("Enter Array size: ");
     scanf("%d", &n);
+    if (n <= 0) /* exit program if arraysize is not greater than 0 */
+    {
+        printf("Array size must be Greater than 0!\n");
+        return 1;
+    }
     a = (int *)malloc(n * sizeof(int));
```

Comment:
Here's a little language lawyer comment: I've heard there's a difference between `n * sizeof(int)` and `sizeof(int) * n`, the first causes the compiler to do the multiplication in `int` type instead of `size_t`, so the result gets cast back and forth between `int` and `size_t`, but I doubt if this is true, modern compilers are probably not that stupid